rotkehlchen/chain/evm/decoding/spark/savings/decoder.py

```diff
@@ -0,0 +1,321 @@
+        """
+        if context.tx_log.topics[0] != SWAP_TOPIC:
+            return DEFAULT_DECODING_OUTPUT
+
+        asset_in = self.base.get_or_create_evm_token(bytes_to_address(context.tx_log.topics[1]))
+        asset_out = self.base.get_or_create_evm_token(bytes_to_address(context.tx_log.topics[2]))
+        amount_in = token_normalized_value_decimals(
+            token_amount=int.from_bytes(context.tx_log.data[32:64]),
+            token_decimals=asset_in.decimals,
+        )
+        amount_out = token_normalized_value_decimals(
+            token_amount=int.from_bytes(context.tx_log.data[64:96]),
+            token_decimals=asset_out.decimals,
+        )
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.amount == amount_in and
+                event.asset == asset_in
+            ):
+                event.counterparty = CPT_SPARK
+                if asset_in.evm_address not in self.spark_savings_tokens:
+                    event.event_type = HistoryEventType.DEPOSIT
+                    event.event_subtype = HistoryEventSubType.DEPOSIT_FOR_WRAPPED
+                    event.notes = f'Deposit {amount_in} {asset_in.symbol} in Spark Savings'
+                else:  # swap exact out
+                    event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                    event.notes = f'Return {amount_in} {asset_in.symbol} into Spark Savings'
+
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.amount == amount_out and
+                event.asset == asset_out
+            ):
+                event.counterparty = CPT_SPARK
+                if asset_out.evm_address in self.spark_savings_tokens:
+                    event.event_type = HistoryEventType.RECEIVE
+                    event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                    event.notes = f'Receive {amount_out} {asset_out.symbol} from depositing into Spark Savings'  # noqa: E501
+                else:  # swap exact out
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.REDEEM_WRAPPED
+                    event.notes = f'Remove {amount_out} {asset_out.symbol} from Spark Savings'
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_spark_tokens_deposit_withdrawal(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes deposit and withdrawal events for Spark tokens (e.g., sUSDC).
+        Similar to PSM deposits/withdrawals but for direct token interactions.
+        """
+        # Check if this is a sky migration event on Ethereum and delegate to the migration decoder
+        if (
+            self.evm_inquirer.chain_id == ChainID.ETHEREUM and
```

Comment:
if this can only happen in ethereum sound like this method should be implemented in the ethereum subclass + the addresses to decoder extended there